crates/nu-command/src/filesystem/ls.rs

```diff
@@ -979,10 +980,16 @@ mod windows_helper {
 #[allow(clippy::type_complexity)]
 fn read_dir(
     f: &Path,
+    use_threads: bool,
 ) -> Result<Box<dyn Iterator<Item = Result<PathBuf, ShellError>> + Send>, ShellError> {
     let iter = f.read_dir()?.map(|d| {
         d.map(|r| r.path())
             .map_err(|e| ShellError::IOError { msg: e.to_string() })
     });
-    Ok(Box::new(iter))
+
+    if use_threads {
+        Ok(Box::new(iter))
+    } else {
+        Ok(Box::new(sorted(iter)))
```

Comment:
Let's collect the iterator and sort it, so we don't need to use `itertools::sorted`.  And no changes is required on `ShellError`.

Reply:
Ok, I think now it should be as you describe it.